TYPO3.Neos/Classes/TYPO3/Neos/TypoScript/DimensionsMenuImplementation.php

```diff
@@ -0,0 +1,216 @@
+            unset($targetDimensionsToMatch[$pinnedDimensionName]);
+        }
+
+        foreach ($this->contentDimensionCombinator->getAllAllowedCombinations() as $allowedCombination) {
+            $targetDimensions = $this->calculateTargetDimensionsForCombination($allowedCombination);
+
+            if ($pinnedDimensionName !== null && is_array($pinnedDimensionValues)) {
+                if (!in_array($targetDimensions[$pinnedDimensionName], $pinnedDimensionValues)) {
+                    continue;
+                }
+            }
+
+            // skip variants not matching the current target dimensions (except the dimension this menu covers)
+            if ($targetDimensionsToMatch !== []) {
+                foreach ($targetDimensionsToMatch as $dimensionName => $dimensionValue) {
+                    if ($targetDimensions[$dimensionName] !== $dimensionValue) {
+                        continue 2;
+                    }
+                }
+            }
+
+            $nodeInDimensions = $this->getNodeInDimensions($allowedCombination, $targetDimensions);
+
+            // no match, so we look further...
+            if ($nodeInDimensions === null && $includeAllPresets) {
+                foreach ($allowedCombination[$pinnedDimensionName] as $allowedPresetIdentifier) {
+                    $acceptableCombination = [$pinnedDimensionName => $allDimensionPresets[$pinnedDimensionName]['presets'][$allowedPresetIdentifier]['values']];
+                    $allowedAdditionalPresets = $this->configurationContentDimensionPresetSource->getAllowedDimensionPresetsAccordingToPreselection('country', [$pinnedDimensionName => $allowedPresetIdentifier]);
+                    foreach ($allowedAdditionalPresets as $allowedAdditionalDimensionName => $allowedAdditionalPreset) {
+                        $acceptableCombination[$allowedAdditionalDimensionName] = $allowedAdditionalPreset['presets'][$allowedAdditionalPreset['defaultPreset']]['values'];
+                    }
+                    $nodeInDimensions = $this->getNodeInDimensions($acceptableCombination, []);
+                    if ($nodeInDimensions !== null) {
+                        continue;
+                    }
+                }
+            }
+
+            if ($nodeInDimensions !== null && $this->isNodeHidden($nodeInDimensions)) {
+                $nodeInDimensions = null;
+            }
+
+            // determine metadata for target dimensions of node
+            array_walk($targetDimensions, function (&$dimensionValue, $dimensionName, $allDimensionPresets) use ($pinnedDimensionName) {
+                $dimensionValue = [
+                    'value' => $dimensionValue,
+                    'label' => $allDimensionPresets[$dimensionName]['presets'][$dimensionValue]['label'],
+                    'isPinnedDimension' => ($pinnedDimensionName === null || $dimensionName == $pinnedDimensionName) ? true : false
+                ];
+            }, $allDimensionPresets);
```

Comment:
Wait how does that even work, since targetDimensions doesn't contain the actual target values anymore? I mean how is it used as this change does not contain any usage.
I added them to use them in a context call which is no longer possible now.